### PR TITLE
tracing nullPointerException fix

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
Addressing the following issue: https://github.com/strimzi/client-examples/issues/85

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,3 +108,5 @@ To enable tracing, configure the Jaeger Tracer using [environment variables](htt
 
 To run Jaeger Tracing, you can also use the provided example in [`deployment-tracing-jaeger.yaml`](./java/kafka/deployment-tracing-jaeger.yaml).
 To run Opentelemetry Tracing, you can also use the provided example in [`deployment-tracing-opentelemetry.yaml`](./java/kafka/deployment-tracing-opentelemetry.yaml).
+
+*NOTE* : Jaeger Tracink eg is only supported up to Kafka version 2.8.0. For Kafka 2.8 +, use OpenTelemetry.
```
Comment (scholzj) [inline README.md]: I do not think this is true and should be here.

File: java/kafka/consumer/src/main/java/KafkaConsumerExample.java [hunk 2]
```diff
@@ -27,43 +27,45 @@ public static void main(String[] args) {
         int receivedMsgs = 0;
 
         TracingSystem tracingSystem = config.getTracingSystem();
-        if (tracingSystem != TracingSystem.NONE) {
-            if (tracingSystem == TracingSystem.JAEGER) {
-                TracingInitializer.jaegerInitialize();
+        if (tracingSystem != null) {
```
Comment (scholzj) [inline java/kafka/consumer/src/main/java/KafkaConsumerExample.java]: When would this be null? Also, why if this is null do we skip the messages? I would expect this to be either NONE or JAEGER or OPENTELEMETRY.

File: java/kafka/consumer/src/main/java/KafkaConsumerExample.java [hunk 3]
```diff
@@ -27,43 +27,45 @@ public static void main(String[] args) {
...
+                } else if (tracingSystem == TracingSystem.OPENTELEMETRY) {
+                    TracingInitializer.otelInitialize();
 
-                props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, io.opentelemetry.instrumentation.kafkaclients.TracingConsumerInterceptor.class.getName());
-            } else {
-                log.error("Error: TRACING_SYSTEM {} is not recognized or supported!", tracingSystem);
+                    props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, io.opentelemetry.instrumentation.kafkaclients.TracingConsumerInterceptor.class.getName());
+                } else {
+                    log.error("Error: TRACING_SYSTEM {} is not recognized or supported!", tracingSystem);
+                }
```
Comment (scholzj) [inline java/kafka/consumer/src/main/java/KafkaConsumerExample.java]: Thsi should be just one if-else -> you should not nest it.

File: java/kafka/pom.xml [hunk 4]
```diff
@@ -18,7 +18,7 @@
         <maven.compiler.target>11</maven.compiler.target>
         <log4j.version>2.17.1</log4j.version>
         <slf4j-simple.version>1.6.2</slf4j-simple.version>
-        <kafka.version>2.8.0</kafka.version>
+        <kafka.version>3.3.1</kafka.version>
```
Comment (scholzj) [inline java/kafka/pom.xml]: You seem to be mixing two different PRs.

File: java/kafka/producer/src/main/java/KafkaProducerExample.java [hunk 5]
```diff
@@ -31,6 +31,7 @@ public static void main(String[] args) throws InterruptedException {
         List<Header> headers = null;
 
         TracingSystem tracingSystem = config.getTracingSystem();
+       // if (TracingSystem == null)
```
Comment (scholzj) [inline java/kafka/producer/src/main/java/KafkaProducerExample.java]: Can be deleted?

**Review by scholzj — APPROVED** [review]
Thanks

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!
